src/entity/Living.php

```diff
@@ -438,6 +444,127 @@ public function setOnFire(int $seconds) : void{
 		parent::setOnFire($seconds - (int) min($seconds, $seconds * $this->getHighestArmorEnchantmentLevel(VanillaEnchantments::FIRE_PROTECTION()) * 0.15));
 	}
 
+	/**
+	 * Returns the number of ticks this entity has accumulated toward being frozen.
+	 */
+	public function getFreezeProgressTicks() : int{
+		return $this->freezeProgressTicks;
+	}
+
+	/**
+	 * Set the entity's freeze progress in ticks.
+	 *
+	 * @throws \InvalidArgumentException if $freezeProgressTicks is negative
+	 */
+	public function setFreezeProgressTicks(int $freezeProgressTicks) : void{
+		if($freezeProgressTicks < 0){
+			throw new \InvalidArgumentException("Freeze ticks cannot be negative");
+		}
+		$this->freezeProgressTicks = $freezeProgressTicks;
+		$this->networkPropertiesDirty = true;
+	}
+
+	/**
+	 * Returns freeze progress as a normalized value between 0.0 and 1.0.
+	 * This is the value sent to clients to control the freezing visual effect.
+	 */
+	public function getFreezeProgressRatio() : float{
+		return min(1.0, $this->freezeProgressTicks / max(1, $this->getFreezeThresholdTicks()));
+	}
+
+	/**
+	 * Returns the number of ticks required for this entity to be considered fully frozen.
+	 */
+	public function getFreezeThresholdTicks() : int{
+		return 140;
+	}
+
+	/**
+	 * Returns whether the entity is currently flagged as accumulating freeze progress (transient).
+	 */
+	public function isAccumulatingFreeze() : bool{
+		return $this->isAccumulatingFreeze;
+	}
+
+	/**
+	 * Sets the transient freezing flag. This flag indicates that the entity should accumulate freeze progress during
+	 * the next run of `entityBaseTick()`.
```

Comment:
This sounds like a lot of words to say `Sets whether the entity is accumulating freeze progress` or something to that effect. (Although I don't much like "accumulating freeze" as I mentioned above.)